### PR TITLE
Added fix for Kubernetes btrfs issue #38337

### PR DESCRIPTION
**Pull request by dmrub**
Please see https://github.com/kubernetes/kubernetes/issues/38337 and https://github.com/kubernetes/kubernetes/pull/39862 for details.

### CODE REVIEW

**Comment by dashpole** [comment]
@k8s-bot ok to test

File: fs/fs.go [hunk 1]
```diff
@@ -149,6 +149,31 @@ func processMounts(mounts []*mount.Info, excludedMountpointPrefixes []string) ma
...
+			if err != nil {
+				glog.Warningf("stat failed on %s with error: %s", mount.Source, err)
+			} else {
+				glog.Infof("btrfs mount %#v", mount)
+				if buf.Mode&syscall.S_IFMT == syscall.S_IFBLK {
+					err := syscall.Stat(mount.Mountpoint, buf)
+					if err != nil {
+						glog.Warningf("stat failed on %s with error: %s", mount.Mountpoint, err)
+					} else {
+						glog.Infof("apply fix to btrfs mount")
```
Comment (dashpole) [inline fs/fs.go]: nit: remove this line.

File: fs/fs.go [hunk 2]
```diff
@@ -149,6 +149,31 @@ func processMounts(mounts []*mount.Info, excludedMountpointPrefixes []string) ma
 			continue
 		}
 
+		// btrfs fix
```
Comment (dashpole) [inline fs/fs.go]: could you expand this comment to explain what this is fixing?

File: fs/fs.go (hunk 1 above)
Comment (dmrub) [inline fs/fs.go]: Which line do you mean ? The empty one ?

Comment (dashpole) [inline fs/fs.go]: The `apply fix to btrfs mount` line.  Doesnt seem like it conveys any important information.

Comment (dmrub) [inline fs/fs.go]: I removed this line and added comment:
https://github.com/dmrub/cadvisor/commit/3e43b4573d5d8ff638b36d5524eacfe6e65215b7

**Comment by dashpole** [comment]
Just linking this here, because it was helpful for me to understand the problem and solution:
https://www.spinics.net/lists/linux-btrfs/msg59039.html

**Pull request merged**